docker/docker.go

```
@@ -1024,11 +1071,36 @@ func (s *DockerPushStep) tagAndPush(imageID string, e *core.NormalizedEmitter, c
 				s.logger.Errorln("Failed to push:", err)
 				return 1, err
 			}
-			s.logger.Println("Pushed container:", s.repository, s.tags)
+			// Covert status messages in stream {...} {...} to a proper json array [{...},{...},...]
+			statusJSON := strings.Join([]string{"[", strings.TrimSuffix(strings.Replace(buf.String(), "\n", ",", -1), ","), "]"}, "")
+			statusMessages := make([]PushStatus, 0)
+			err = json.Unmarshal(([]byte)(statusJSON), &statusMessages)
+			if err != nil {
+				s.logger.Errorln("Failed to parse status outputs from docker push:", err)
+				return 1, err
+			}
+			isContainerPushed := false
+			for _, statusMessage := range statusMessages {
+				if len(strings.TrimSpace(statusMessage.Error)) != 0 {
```

How certain are you that this is a fatal error? Ie. isn't it possible that Docker retries or that the error can be ignored?

---

These status messages are from docker push and this particular error was if we are unable to decode the json message coming out of docker - these messages are after docker have done all retries - so in a way was fatal because if we are unable to decode a message - in forthcoming lines we will not be able to identify if a push was successful. But since I check for presence of sha anyway - I have removed the "fatality" of this decode error but not returning from here in a error condition